internal/k8s/controller_test.go

```diff
@@ -470,79 +472,516 @@ func TestFormatWarningsMessages(t *testing.T) {
+	}{
+		{
+			desc:       "duplicate endpoints in an endpointslice",
+			targetPort: 8080,
+			expectedEndpoints: []podEndpoint{
+				{
+					Address: "1.2.3.4:8080",
+				},
+			},
+			svcEndpointSlices: []discovery_v1.EndpointSlice{
+				{
+					Ports: []discovery_v1.EndpointPort{
+						{
+							Port: &endpointPort,
+						},
+					},
+					Endpoints: []discovery_v1.Endpoint{
+						{
+							Addresses: []string{
+								"1.2.3.4",
+							},
+						},
+						{
+							Addresses: []string{
+								"1.2.3.4",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc:       "two different endpoints in one endpoint slice",
+			targetPort: 8080,
+			expectedEndpoints: []podEndpoint{
+				{
+					Address: "1.2.3.4:8080",
+				},
+				{
+					Address: "5.6.7.8:8080",
+				},
+			},
+			svcEndpointSlices: []discovery_v1.EndpointSlice{
+				{
+					Ports: []discovery_v1.EndpointPort{
+						{
+							Port: &endpointPort,
+						},
+					},
+					Endpoints: []discovery_v1.Endpoint{
+						{
+							Addresses: []string{
+								"1.2.3.4",
+							},
+						},
+						{
+							Addresses: []string{
+								"5.6.7.8",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc:       "duplicate endpoints across two endpointslices",
+			targetPort: 8080,
+			expectedEndpoints: []podEndpoint{
+				{
+					Address: "1.2.3.4:8080",
+				},
+				{
+					Address: "5.6.7.8:8080",
+				},
+				{
+					Address: "10.0.0.1:8080",
+				},
+			},
+			svcEndpointSlices: []discovery_v1.EndpointSlice{
+				{
+					Ports: []discovery_v1.EndpointPort{
+						{
+							Port: &endpointPort,
+						},
+					},
+					Endpoints: []discovery_v1.Endpoint{
+						{
+							Addresses: []string{
+								"1.2.3.4",
+							},
+						},
+						{
+							Addresses: []string{
+								"5.6.7.8",
+							},
+						},
+					},
+				},
+				{
+					Ports: []discovery_v1.EndpointPort{
+						{
+							Port: &endpointPort,
+						},
+					},
+					Endpoints: []discovery_v1.Endpoint{
+						{
+							Addresses: []string{
+								"1.2.3.4",
+							},
+						},
+						{
+							Addresses: []string{
+								"10.0.0.1",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc:              "no endpoints found",
+			targetPort:        8080,
+			expectedEndpoints: nil,
+			svcEndpointSlices: []discovery_v1.EndpointSlice{},
+		},
+	}
+
+	svc := api_v1.Service{
+		TypeMeta: meta_v1.TypeMeta{},
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      "coffee-svc",
+			Namespace: "default",
+		},
+		Spec: api_v1.ServiceSpec{
+			Ports: []api_v1.ServicePort{
+				{
+					Name:       "foo",
+					Port:       80,
+					TargetPort: intstr.FromInt(8080),
+				},
+			},
+		},
+		Status: api_v1.ServiceStatus{},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			gotEndpoints, _ := lbc.getEndpointsForPortFromEndpointSlices(test.svcEndpointSlices, backendServicePort, &svc)
+
+			if result := unorderedEqual(gotEndpoints, test.expectedEndpoints); !result {
```

Comment:
would the `cmp`  package be helpful here?
```go
if !cmp.Equal(gotEndpoints, test.expectedEndpoints) {
    t.Error(cmp.Diff(gotEndpoints, test.expectedEndpoints))
}
```

Reply:
I tried using `cmp.Diff` and noticed that the test would fail if the value of `gotEndpoint` was not in the same order. On some runs I would get a result like this:

```
lbc.getEndpointsForPortFromEndpointSlices() 
 got [{10.0.0.1:8080  { }} {1.2.3.4:8080  { }} {5.6.7.8:8080  { }}], 
 want [{1.2.3.4:8080  { }} {5.6.7.8:8080  { }} {10.0.0.1:8080  { }}]
```